android/KMEA/app/src/main/java/com/tavultesoft/kmea/KMKeyboard.java

```diff
@@ -1056,8 +1032,22 @@ private String getSubkeyText(String keyID, String keyText) {
    * @param font String font JSON object as a string
    * @return String of modified font information with full paths. If font is invalid, return "''"
    */
-  private String makeFontPaths(String font) {
+  private JSONObject makeFontPaths(String font) {
+
+    if(font == null) {
+      return null;
+    }
+
     try {
+      if (FileUtils.hasFontExtension(font)) {
+        JSONObject jfont = new JSONObject();
+        jfont.put(KMManager.KMKey_FontFamily, font.substring(0, font.length()-4));
```

Comment:
Do we get `IndexOutOfBoundsException` if font.length() < 4?

Reply:
No, because `hasFontExtension` cannot return `true` in that case (i.e. `.ttf` has length 4). It's a little bit assumed but it's safe.